PowerEditor/src/ScitillaComponent/Buffer.cpp

```diff
@@ -957,47 +922,8 @@ bool FileManager::backupCurrentBuffer()
 	return result;
 }
 
-class EventReset final
-{
-public:
-	explicit EventReset(HANDLE h)
-	{
-		_h = h;
-	}
-
-	~EventReset()
-	{
-		::SetEvent(_h);
-		::CloseHandle(_h);
-	}
-
-private:
-	HANDLE _h;
-};
-
 bool FileManager::deleteBufferBackup(BufferID id)
 {
-	HANDLE writeEvent = ::OpenEvent(EVENT_ALL_ACCESS, TRUE, TEXT("nppWrittingEvent"));
-	if (!writeEvent)
-	{
-		// no thread yet, create a event with non-signaled, to block all threads
-		writeEvent = ::CreateEvent(NULL, TRUE, FALSE, TEXT("nppWrittingEvent"));
-	}
-	else
-	{
-		if (::WaitForSingleObject(writeEvent, INFINITE) != WAIT_OBJECT_0)
-		{
-			// problem!!!
-			printStr(TEXT("WaitForSingleObject problem in deleteBufferBackup()!"));
-			return false;
-		}
-
-		// unlocled here, set to non-signaled state, to block all threads
-		::ResetEvent(writeEvent);
-	}
-
-	EventReset reset(writeEvent); // Will reset event in destructor.
-

```

Comment:
No `LongRunningOperation op;` instruction here?

Reply:
You mean in `FileManager::deleteBufferBackup()` ?
It won't hurt to add `LongRunningOperation op` but strictly speaking it is not needed at all. All buffer operations are now channeled through a single thread so no need for locking at all.
I can either remove all `LongRunningOperation` locks from `FileManager` in this file or add a lock in `deleteBufferBackup()` for consistency. What do you prefer?

Reply:
> All buffer operations are now channeled through a single thread

Could you point me this part?

Reply:
Yes, at the very beginning of this Git diff I have changed the direct call to `MainFileManager->backupCurrentBuffer()` to `::PostMessage(Notepad_plus_Window::gNppHWND, NPPM_INTERNAL_SAVEBACKUP, 0, 0);`.
This way the actual buffer backup will go through the message queue which is serialized - no concurrency.
The other operations here were not synchronized (locked) anyway but they also are always triggered from the GUI thread (again through the message queue). Am I missing anything?

Reply:
Well played!